src/index.js

```diff
@@ -25,94 +25,112 @@ const path = require('path')
 const request = require('request')
 const tarFS = require('tar-fs')
 const unzip = require('unzip-stream')
-const support = require('./check-support')
 const pkg = require('./../package.json')
 
-// Check package.json for default config
-const goIpfsInfo = pkg['go-ipfs']
-
-const goIpfsVersion = (goIpfsInfo && goIpfsInfo.version)
-  ? pkg['go-ipfs'].version
-  : 'v' + pkg.version.replace(/-[0-9]+/, '')
-
-let distUrl = (goIpfsInfo && goIpfsInfo.distUrl)
-  ? pkg['go-ipfs'].distUrl
-  : 'https://dist.ipfs.io'
-
-// Main function
-function download (version, platform, arch, installPath) {
+function unpack ({ url, installPath, stream }) {
   return new Promise((resolve, reject) => {
-    //            Environment Variables           Args        Defaults
-    version = process.env.TARGET_VERSION || version || goIpfsVersion
-    platform = process.env.TARGET_OS || platform || goenv.GOOS
-    arch = process.env.TARGET_ARCH || arch || goenv.GOARCH
-    distUrl = process.env.GO_IPFS_DIST_URL || distUrl
-    installPath = installPath ? path.resolve(installPath) : path.resolve(process.cwd())
-
-    // Make sure we support the requested package
-    try {
-      support.verify(version, platform, arch)
-    } catch (e) {
-      return reject(e)
+    if (url.endsWith('.zip')) {
+      return stream.pipe(
+        unzip
+          .Extract({ path: installPath })
+          .on('close', resolve)
+          .on('error', reject)
+      )
     }
 
-    // Flag for Windows
-    const isWindows = support.isWindows(platform)
-
-    // Create the download url
-    const fileExtension = isWindows ? '.zip' : '.tar.gz'
-    const fileName = 'ipfs_' + version + '_' + platform + '-' + arch + fileExtension
-    const url = distUrl + '/go-ipfs/' + version + '/go-' + fileName
+    return stream
+      .pipe(gunzip())
+      .pipe(
+        tarFS
+          .extract(installPath)
+          .on('finish', resolve)
+          .on('error', reject)
+      )
+  })
+}
 
-    // Success callback wrapper
-    // go-ipfs contents are in 'go-ipfs/', so append that to the path
-    const done = () => resolve({
-      fileName: fileName,
-      installPath: path.join(installPath, '/go-ipfs/')
+function download ({ installPath, url }) {
+  return new Promise((resolve, reject) => {
+    request.get(url, (err, res, body) => {
```

Comment:
Might be easier to use `node-fetch` instead.

Reply:
I just kept request to use with the `unpack` function.

Reply:
Not sure I follow? I don't think `unpack` requires request does it? If you just need a stream you can pass it `res.body` from `node-fetch`. See https://www.npmjs.com/package/node-fetch#streams

```js
function download ({ installPath, url }) {
  const res = await fetch(url)
  if (!res.ok) throw new Error(`unexpected status ${res.status}`)
  return unpack({ url, installPath, stream: res.body })
}
```

...right?

Reply:
Forget forget 😆 Done!